Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenu.cpp

```diff
@@ -21,8 +21,9 @@ namespace AzToolsFramework
     {
     }
 
-    EditorMenu::EditorMenu(const AZStd::string& name)
-        : m_menu(new QMenu(name.c_str()))
+    EditorMenu::EditorMenu(const AZStd::string& identifier, const AZStd::string& name)
+        : m_identifier(identifier)
+        , m_menu(new QMenu(name.c_str()))
```

Comment:
suggestion: While we're here, it might be nice to update `const AZStd::string& identifier` to `AZStd::string identifier` so that we can move the value into `m_identifier`. This is so if we pass a temporary to the constructor, we won't end up making two copies. The code also more clearly conveys intent.

```c++
EditorMenu::EditorMenu(AZStd::string identifier, const AZStd::string& name)
        : m_identifier(AZStd::move(identifier))
        , m_menu(new QMenu(name.c_str()))
```

speculative: It might also be worth changing `AZStd::string& name` to either a `AZStd::string_view` or `const char*` to stop having to construct a full `AZStd::string` and then throw it away when we create the `QString` for the QMenu constructor. It might not be worth it in this case but I just wanted to mention it in case.

Reply:
Brilliant thank you! 🙂